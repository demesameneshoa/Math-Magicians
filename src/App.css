body {
  height: 100vh;
  background: #ddd;
  display: flex;
  padding: 2rem;
  font-family: 'Roboto', sans-serif;
}

.App {
  margin: 0;
  padding: 0;
  width: max-content;
  box-sizing: border-box;
  display: flex;
  text-align: center;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.quote-container {
  max-width: 22rem;
  margin-top: 1rem;
}

.quote {
  font-style: oblique;
}

.Display {
  height: 4rem;
  width: 100%;
  background-color: gray;
  color: white;
  display: flex;
  font-size: larger;
  font-weight: bold;
  justify-content: flex-end;
  align-items: center;
  padding: 0;
  margin: 0;
  border-radius: 1px;
}

.ButtonPanel {
  display: flex;
  flex-direction: column;
}

.button {
  width: 70px;
  height: 70px;
}

.operation-button {
  background: rgb(230, 164, 43);
}

.LastRow {
  display: grid;
  grid-template-columns: 2fr 1fr 1fr;
}

.LastRow button {
  width: 100%;
}

.op button {
  background-color: rebeccapurple;
}

ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  overflow: hidden;
  background-color: #aaa9a9;
}

li {
  float: left;
}

li a {
  display: block;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

/* Change the link color to #111 (black) on hover */
li a:hover {
  background-color: #111;
}