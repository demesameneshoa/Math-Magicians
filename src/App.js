import Calculator from './components/Calculator';
import Quote from './components/Quotes';
import './App.css';

function App() {
  return (
    <>
      <div className="Calculator-container">
        <Calculator />
      </div>
      <div className="quote-container">
        <Quote />
      </div>
    </>
  );
}

export default App;
