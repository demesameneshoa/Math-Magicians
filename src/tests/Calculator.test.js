import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../components/Calculator';

describe('Calculator component unit test', () => {
  test('Calculator Component Renders Properly ', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('The Calculator Component Contains The Display Component ', () => {
    render(<App />);
    expect(screen.getByTestId('header')).toBeInTheDocument();
  });
});
