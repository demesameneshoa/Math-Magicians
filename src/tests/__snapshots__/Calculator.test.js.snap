// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component unit test Calculator Component Renders Properly  1`] = `
<div
  className="wrapper"
  data-testid="header"
>
  <div>
    <h2>
      Let's Do Some Math
    </h2>
  </div>
  <div
    className="App"
  >
    <div
      className="Display"
    >
      0
    </div>
    <div
      className="ButtonPanel"
    >
      <div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button operation-button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button operation-button"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button operation-button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button operation-button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="LastRow"
      >
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button operation-button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
